src/Tools/dotnet-monitor/Egress/Extension/EgressExtension.cs

```diff
@@ -70,10 +70,40 @@ public void AddEnvironmentVariable(string key, string value)
             return EgressArtifact(payload, action, token);
         }
 
+        /// <inheritdoc/>
+        public async Task<EgressArtifactResult> ValidateProviderAsync(
+            string providerName,
+            EgressArtifactSettings settings,
+            Func<Stream, CancellationToken, Task> action,
+            CancellationToken token)
+        {
+            EgressArtifactResult result = new();
+
+            if (!_manifest.Modes.Any())
+            {
+                result.Succeeded = true;
+            }
+            else if (_manifest.Modes.Contains(ExtensionModes.Validate))
+            {
+                ExtensionEgressPayload payload = new()
+                {
+                    Settings = settings,
+                    Configuration = GetConfigurationSection(providerName, _manifest.Name),
+                    Properties = _configurationProvider.GetAllProperties(),
+                    ProviderName = providerName,
+                };
+
+                result = await EgressArtifact(payload, action, token, ExtensionModes.Validate);
+            }
+
+            return result;
+        }
+
         public async Task<EgressArtifactResult> EgressArtifact(
             ExtensionEgressPayload payload,
             Func<Stream, CancellationToken, Task> action,
-            CancellationToken token)
+            CancellationToken token,
+            ExtensionModes mode = ExtensionModes.Execute)
```

Comment:
nit: Remove the default value and have the caller be explicit about which mode to use.